programs/voter-stake-registry/src/state/deposit_entry.rs

```diff
@@ -204,23 +211,30 @@ impl DepositEntry {
         //                      = q * (q - 1) / 2
         //
 
-        // In the example above, periods_total was 5.
-        let denominator = periods_total.checked_mul(lockup_saturation_secs).unwrap();
-
         let secs_to_closest_cliff = self
             .lockup
             .seconds_left(curr_ts)
-            .checked_sub(period_secs.checked_mul(periods_left - 1).unwrap())
+            .checked_sub(
+                period_secs
+                    .checked_mul(periods_left.saturating_sub(1))
+                    .unwrap(),
+            )
             .unwrap();
 
-        let lockup_saturation_periods =
-            (lockup_saturation_secs + period_secs - secs_to_closest_cliff) / period_secs;
+        if secs_to_closest_cliff >= lockup_saturation_secs {
+            return Ok(max_locked_vote_weight);
+        }
+
+        // In the example above, periods_total was 5.
+        let denominator = periods_total.checked_mul(lockup_saturation_secs).unwrap();
+
+        let lockup_saturation_periods = (lockup_saturation_secs
+            .saturating_sub(secs_to_closest_cliff)
+            .checked_add(period_secs)
+            .unwrap())
+            / period_secs;
```

Comment:
checked_div instead of / ?

Reply:
as I understand it, that only guards against `period_secs == 0`... but you know, why not

Reply:
done